velox/functions/prestosql/window/tests/SimpleAggregatesTest.cpp

```diff
@@ -30,118 +30,96 @@ static const std::vector<std::string> kAggregateFunctions = {
+                       public testing::WithParamInterface<WindowTestParam> {
  public:
-  MultiAggregatesTest() : SimpleAggregatesTest(GetParam()) {}
+  AggregatesTest() : WindowFunctionTest(GetParam()) {}
 };
 
-TEST_P(MultiAggregatesTest, basic) {
-  SimpleAggregatesTest::testWindowFunction(
-      {makeBasicVectors(10)}, kBasicOverClauses);
+// Tests function with a dataset with uniform partitions.
+TEST_P(AggregatesTest, basic) {
+  testWindowFunction({makeSimpleVector(10)});
 }
 
-TEST_P(MultiAggregatesTest, basicWithSortOrders) {
-  SimpleAggregatesTest::testWindowFunction(
-      {makeBasicVectors(10)}, kSortOrderBasedOverClauses);
+// Tests function with a dataset with a single partition containing all the
+// rows.
+TEST_P(AggregatesTest, singlePartition) {
+  testWindowFunction({makeSinglePartitionVector(100)});
 }
 
-TEST_P(MultiAggregatesTest, singlePartition) {
-  // Test all input rows in a single partition.
-  SimpleAggregatesTest::testWindowFunction(
-      {makeSinglePartitionVector(100)}, kBasicOverClauses);
+// Tests function with a dataset with a single partition but 2 input row
+// vectors.
+TEST_P(AggregatesTest, multiInput) {
+  testWindowFunction(
+      {makeSinglePartitionVector(250), makeSinglePartitionVector(50)});
 }
 
-TEST_P(MultiAggregatesTest, singlePartitionWithSortOrders) {
-  SimpleAggregatesTest::testWindowFunction(
-      {makeSinglePartitionVector(100)}, kSortOrderBasedOverClauses);
+// Tests function with a dataset where all partitions have a single row.
+TEST_P(AggregatesTest, singleRowPartitions) {
+  testWindowFunction({makeSingleRowPartitionsVector(50)});
 }
 
-TEST_P(MultiAggregatesTest, randomInput) {
-  auto size = 50;
-  std::vector<VectorPtr> vectors;
-  // Columns c1, c2 are used for k PRECEDING/FOLLOWING frame bounds. So they
-  // should have values >= 1.
-  auto input = makeRowVector(
-      {makeFlatFuzzVector(BIGINT(), size),
-       makeFlatVector<int64_t>(size, [](auto row) { return row % 5 + 1; }),
-       makeFlatVector<int64_t>(size, [](auto row) { return row + 1; }),
-       makeFlatFuzzVector(INTEGER(), size),
-       makeFlatFuzzVector(BIGINT(), size)});
-
-  createDuckDbTable({input});
-  // Add columns c3, c4 in sort order in overClauses to impose a deterministic
-  // output row order in the tests.
-  testWindowFunction(
-      {input}, addSuffixToClauses(", c3, c4", kBasicOverClauses));
+// Tests function with a randomly generated input dataset.
+TEST_P(AggregatesTest, randomInput) {
+  testWindowFunction({makeRandomInputVector(50)});
 }
 
+// Instantiate all the above tests for each combination of aggregate function
+// and over clause.
 VELOX_INSTANTIATE_TEST_SUITE_P(
-    SimpleAggregatesTest,
-    MultiAggregatesTest,
-    testing::ValuesIn(getTestParams(kFrameClauses)));
+    AggregatesTestInstantiation,
+    AggregatesTest,
+    testing::ValuesIn(getWindowTestParams(kAggregateFunctions, kOverClauses)));
 
 class StringAggregatesTest : public WindowTestBase {};
```

Comment:
Why do we need a separate test for strings?

Reply:
The use of String with aggregates exercises the use of the HashStringAllocator passed to the AggregateWindow.